lib/collection/src/collection/distance_matrix.rs

```diff
@@ -0,0 +1,142 @@
+use std::collections::HashSet;
+use std::time::Duration;
+
+use rand::prelude::SliceRandom;
+use segment::data_types::vectors::{
+    NamedMultiDenseVector, NamedVectorStruct, VectorStructInternal,
+};
+use segment::types::{Filter, HasIdCondition, PointIdType, ScoredPoint, WithVector};
+
+use crate::collection::Collection;
+use crate::operations::query_enum::QueryEnum;
+use crate::operations::shard_selector_internal::ShardSelectorInternal;
+use crate::operations::types::{
+    CollectionResult, CoreSearchRequest, CoreSearchRequestBatch, PointRequestInternal,
+};
+
+#[derive(Debug, Default)]
+struct DistanceMatrixResult {
```

Comment:
nit: let's use `*Response` for success response types to not confuse with `*Result` which can have an `Err` variant